hack/tests/e2e-helm-new-shell.sh

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
+
+set -eux
+
+source hack/lib/test_lib.sh
+source hack/lib/image_lib.sh
+
+DEST_IMAGE="quay.io/example/nginx-operator:v0.0.2"
+TMPDIR="$(mktemp -d)"
+trap_add 'rm -rf $TMPDIR' EXIT
+
+test_namespace="nginx-operator-system"
+
+remove_operator() {
+    make undeploy
+    kubectl delete --ignore-not-found namespace ${test_namespace}
+}
+
+test_operator() {
+    # kind has an issue with certain image registries (ex. redhat's), so use a
+    # different test pod image.
+    local metrics_test_image="fedora:latest"
+
+    # wait for operator pod to run
+    if ! timeout 1m kubectl rollout status deployment.apps/nginx-operator-controller-manager -n ${test_namespace} ;
+    then
+        kubectl get events --namespace=${test_namespace}
+        kubectl logs deployment.apps/nginx-operator-controller-manager -c manager --namespace=${test_namespace}
+        exit 1
+    fi
+
```

Comment:
This test is missing the checks that `curl` the metrics endpoints that currently exists in the legacy shell-based test. We need to check all the same scenarios to make sure we don't accidentally break something in the transition.

Reply:
But we will not remove the metrics? Actually, after re-check this one could we not to remove the service monitor checks indeed?